runtime/lua/vim/pos.lua

```diff
@@ -172,12 +172,41 @@ function Pos.lsp(buf, pos, position_encoding)
   -- When on the first character,
   -- we can ignore the difference between byte and character.
   if col > 0 then
-    col = vim.str_byteindex(get_line(buf, row), position_encoding, col)
+    -- `strict_indexing` is disabled, because LSP responses are asynchronous,
+    -- and the buffer content may have changed, causing out-of-bounds errors.
+    col = vim.str_byteindex(get_line(buf, row), position_encoding, col, false)
   end
 
   return Pos.new(row, col, { buf = buf })
 end
 
+--- Converts |vim.Pos| to cursor position.
+---@param pos vim.Pos
+---@return [integer, integer]
+function Pos.to_cursor(pos)
+  return { pos.row + 1, pos.col }
+end
+
+--- Creates a new |vim.Pos| from cursor position.
+---@param pos [integer, integer]
+function Pos.cursor(pos)
+  return Pos.new(pos[1] - 1, pos[2])
+end
+
+--- Converts |vim.Pos| to extmark position.
+---@param pos vim.Pos
+---@return [integer, integer]
+function Pos.to_extmark(pos)
```

Comment:
nit/off-topic: as a cheap hint in this source code, maybe it makes sense to name the parameter `self` for the `to_x` cases?
```suggestion
function Pos.to_extmark(self)
```

Reply:
To implement such a hint, a more concise way is to use `function Pos:to_extmark()`, which also saves filling in parameters, but I tend not to do this because we don’t really want to introduce a lot of OOP for this lightweight type. These APIs are just functions that can be called in an OOP style.